openpype/pipeline/file_collection.py

```diff
@@ -0,0 +1,292 @@
+"""Utility functions for file sequences and collections
+"""
+import os
+import re
+import clique
+from openpype.lib import Logger
+from openpype.pipeline import frame_utils
+
+
+def collect_filepaths_from_sequential_path(
+    file_path,
+    frame_start=None,
+    frame_end=None,
+    only_existing=False,
+):
+    """Generate expected files from path
+
+    Args:
+        file_path (str): Absolute path with any sequential pattern (##, %02d)
+        frame_start (Optional[int]): Start frame of the sequence
+        frame_end (Optional[int]): End frame of the sequence
+        only_existing (Optional[bool]): Ensure that files exists.
+
+    Returns:
+        Any[list[str], str]: List of absolute paths to files
+            (frames of a sequence) or path if single file
+    """
+
+    dirpath = os.path.normpath(
+        os.path.dirname(file_path))
+
+    filename = os.path.basename(file_path)
+
+    formattable_string = convert_filename_to_formattable_string(
```

Comment:
Wouldn't we be able to achieve something like what this function is doing by using `clique.assemble` ?